src/Plugin/rest/resource/WebformAllFormSubmissions.php

```diff
@@ -106,53 +109,91 @@ public function get(string $webform_id): ModifiedResourceResponse {
+
+      return new ModifiedResourceResponse($errors, Response::HTTP_INTERNAL_SERVER_ERROR);
+    }
+
     // Query for webform submissions with this webform_id.
-    $query = $this->entityTypeManager->getStorage('webform_submission')->getQuery()
+    $submissionQuery = $submissionEntityStorage->getQuery()
       ->condition('webform_id', $webform_id);
 
-    $startTimestamp = $this->currentRequest->query->get('starttime');
-    if (is_numeric($startTimestamp)) {
-      $query->condition('created', $startTimestamp, '>=');
-      $result['starttime'] = $startTimestamp;
-    }
+    foreach (self::ALLOWED_DATETIME_QUERY_PARAMS as $param => $operator) {
+      $errors = $this->updateSubmissionQuery($this->currentRequest, $submissionQuery, $param, $operator, $result);
 
-    $endTimestamp = $this->currentRequest->query->get('endtime');
-    if (is_numeric($endTimestamp)) {
-      $query->condition('created', $endTimestamp, '<=');
-      $result['endtime'] = $endTimestamp;
+      if (isset($errors['error'])) {
+        return new ModifiedResourceResponse($errors, Response::HTTP_BAD_REQUEST);
+      }
     }
 
-    $query->accessCheck(FALSE);
-    $sids = $query->execute();
+    // Complete query.
+    $submissionQuery->accessCheck(FALSE);
+    $sids = $submissionQuery->execute();
 
     $submissionData = [];
 
-    foreach ($sids as $sid) {
-      /** @var \Drupal\webform\WebformSubmissionInterface $webform_submission */
-      $webform_submission = $this->entityTypeManager->getStorage('webform_submission')->load($sid);
-
-      $url = Url::fromRoute(
-            'rest.webform_rest_submission.GET',
-            [
-              'webform_id' => $webform_id,
-              'uuid' => $webform_submission->uuid(),
-            ],
-            [
-              'absolute' => TRUE,
-            ]
-        )->toString();
-
-      $submissionData[$sid] = $url;
+    if (!empty($sids)) {
+      $submissions = $submissionEntityStorage->loadMultiple($sids);
+
+      foreach ($submissions as $submission) {
+        $url = Url::fromRoute(
+          'rest.webform_rest_submission.GET',
+          [
+            'webform_id' => $webform_id,
+            'uuid' => $submission->uuid(),
+          ]
+        )
+          ->setAbsolute()
+          ->toString(TRUE)->getGeneratedUrl();
+
+        $submissionData[$submission->id()] = $url;
+      }
     }
 
     $result['submissions'] = $submissionData;
 
     return new ModifiedResourceResponse($result);
   }
 
+  /**
+   * Updates submission query with request query parameters.
+   *
+   * @phpstan-param array<string, mixed> $result
+   * @phpstan-return array<string, mixed>
+   */
+  private function updateSubmissionQuery(Request $request, QueryInterface $submissionQuery, string $parameter, string $operator, array &$result): array {
+    // Handle starttime request query.
+    $timeQuery = $request->query->get($parameter);
+
+    if (!empty($timeQuery)) {
+      try {
+        $startTime = new \DateTime($timeQuery);
```

Comment:
```suggestion
        $startTime = new \DateTimeImmutable($timeQuery);
```

(cf. https://www.php.net/manual/en/datetime.construct.php)